xla/service/gpu/cublas_cudnn.cc

```diff
@@ -181,7 +254,128 @@ std::string CudnnfMHAKindToString(CudnnfMHAKind kind) {
       return "fmha_bias_softmax_with_dropout";
     case CudnnfMHAKind::kScaleBiasSoftmax:
       return "fmha_bias_softmax";
+    // backward
+    case CudnnfMHAKind::kBackwardBmmBmm:
+      return "fused_batched_matmuls_backward";
+    case CudnnfMHAKind::kBackwardSoftmax:
+      return "fmha_softmax_backward";
+    case CudnnfMHAKind::kBackwardSoftmaxDropout:
+      return "fmha_softmax_with_dropout_backward";
+    case CudnnfMHAKind::kBackwardScaleMaskSoftmax:
+      return "fmha_scaled_masked_softmax_backward";
+    case CudnnfMHAKind::kBackwardScaleMaskSoftmaxDropout:
+      return "fmha_scaled_masked_softmax_with_dropout_backward";
+    case CudnnfMHAKind::kBackwardScaleBiasMaskSoftmax:
+      return "fmha_scaled_bias_masked_softmax_backward";
+    case CudnnfMHAKind::kBackwardScaleBiasMaskSoftmaxDropout:
+      return "fmha_scaled_bias_masked_softmax_with_dropout_backward";
+    case CudnnfMHAKind::kBackwardScaleBiasSoftmaxDropout:
+      return "fmha_bias_softmax_with_dropout_backward";
+    case CudnnfMHAKind::kBackwardScaleBiasSoftmax:
+      return "fmha_bias_softmax_backward";
+  }
+}
+
+// Give fmha instruction a more useful name than "custom-call.42".
+Status SetFMHAInstructionName(HloModule* module, HloInstruction* fmha) {
+  if (fmha->custom_call_target() == kCudnnfMHABmmBmmCallTarget) {
```

Comment:
Maybe unify all these branches? Like if the call target has a certain prefix, call SetAndUniquifyInstrName on the custom_call_target()?

Reply:
i will add a switch to return prefix based on call target and call SetAndUniquifyInstrName